sdk/batch/azure-resourcemanager-batch/src/test/java/com/azure/resourcemanager/batch/BatchTests.java

```diff
@@ -28,56 +45,232 @@
+        String displayName = "badn" + randomPadding();
+        Application application = batchManager
+            .applications()
+            .define(applicationName)
+            .withExistingBatchAccount(resourceGroup, batchAccountName)
+            .withDisplayName(displayName)
+            .withAllowUpdates(true)
+            .create();
+        Assertions.assertEquals(application.displayName(), displayName);
+        Assertions.assertEquals(application.name(), applicationName);
+        Assertions.assertNull(application.defaultVersion());
+
+        // update application
+        String newDisplayName = "newbadn" + randomPadding();
+        application
+            .update()
+            .withDisplayName(newDisplayName)
+            .apply();
+        Assertions.assertNotEquals(displayName, application.displayName());
+
+        // default package version not defined yet
+        String packageVersion = "version" + randomPadding();
+        Assertions.assertThrows(
+            Exception.class,
+            () -> application
+                .update()
+                .withDefaultVersion(packageVersion)
+                .apply()
+        );
+
+        ApplicationPackage applicationPackage = batchManager
+            .applicationPackages()
+            .define(packageVersion)
+            .withExistingApplication(resourceGroup, batchAccountName, applicationName)
+            .create();
+        Assertions.assertNotNull(applicationPackage);
+        Assertions.assertNull(applicationPackage.lastActivationTime());
+
+        // delete
+        // all application packages must be deleted before the application can be deleted
+        batchManager.applicationPackages().delete(resourceGroup, batchAccountName, applicationName, packageVersion);
+        batchManager.applications().delete(resourceGroup, batchAccountName, applicationName);
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testCRUDBatchPool() {
+        // batch account
+        final String batchAccountName = "sa" + randomPadding();
+        BatchAccount account = batchManager
+            .batchAccounts()
+            .define(batchAccountName)
+            .withRegion(REGION)
+            .withExistingResourceGroup(resourceGroup)
+            .create();
+        // batch pool create
+        String poolName = "bp" + randomPadding();
+        String poolDisplayName = "bpdn" + randomPadding();
+        Pool pool = batchManager.pools()
+            .define(poolName)
+            .withExistingBatchAccount(resourceGroup, batchAccountName)
+            .withDisplayName(poolDisplayName)
+            .withDeploymentConfiguration(
+                new DeploymentConfiguration()
+                    .withCloudServiceConfiguration(
+                        new CloudServiceConfiguration()
+                            .withOsFamily("4")
+                            .withOsVersion("WA-GUEST-OS-4.45_201708-01")))
```

Comment:
Do we have alternative to these configure?
Would like to avoid batch someday drop this OsVersion.

Reply:
OK, OS version can be omitted in the configuration.